### ISSUE TITLE
[Linux] process IO wait

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#1527]
I'm benchmarking a network app by using 2 different algorithms: plain send() and zero-copy sendfile(). I would like to know how much CPU time the process spends waiting for IO to complete (e.g. on a single send() or sendfile() call). Linux exposes this information at system level (`psutil.cpu_times().iowait`). It seems we can do this also on a per-process level by reading `delayacct_blkio_ticks` field from /proc/pid/stat:
https://unix.stackexchange.com/questions/459288/where-to-find-iowait-stat-for-a-single-process-as-a-file
I still have to understand whether this is really useful to have and whether other platforms provide the same metric (I guess they don't), so basically this is a possible enhancement proposal. The API should be exposed as a `Process.iowait()` method.

**Comment by giampaolo** [comment]
...or possibly it can be added to the namedtuple returned by `Process.io_counters()`.

**Comment by giampaolo** [comment]
On a second thought it's better to return this as part of Process `cpu_times()` (it also requires less privileges). Attaching a patch here which also includes process CPU guest times. I wrote a script which keeps writing stuff in a file though, and `psutil.cpu_times().iowait` increases quite a bit while `psutil.Process().cpu_times().iowait` does much more rarely so I'm not sure how to interpret this.

```diff
diff --git a/psutil/_pslinux.py b/psutil/_pslinux.py
index e4bc7d75..4bb549e2 100644
--- a/psutil/_pslinux.py
+++ b/psutil/_pslinux.py
@@ -200,6 +200,10 @@ pmmap_ext = namedtuple(
 pio = namedtuple('pio', ['read_count', 'write_count',
                          'read_bytes', 'write_bytes',
                          'read_chars', 'write_chars'])
+# psutil.Process.cpu_times()
+pcputimes = namedtuple('pcputimes',
+                       ['user', 'system', 'children_user', 'children_system',
+                        'guest', 'children_guest', 'iowait'])
 
 
 # =====================================================================
@@ -300,6 +304,13 @@ def cat(fname, fallback=_DEFAULT, binary=True):
             raise
 
 
+def listget(ls, idx, default):
+    try:
+        return ls[idx]
+    except IndexError:
+        return default
+
+
 try:
     set_scputimes_ntuple("/proc")
 except Exception:
@@ -1072,7 +1083,6 @@ def disk_io_counters(perdisk=False):
             lines = f.readlines()
         for line in lines:
             fields = line.split()
-            flen = len(fields)
             if flen == 15:
                 # Linux 2.4
                 name = fields[3]
@@ -1576,6 +1586,9 @@ class Process(object):
         ret['children_stime'] = fields[14]
         ret['create_time'] = fields[19]
         ret['cpu_num'] = fields[36]
+        ret['delayacct_blkio_ticks'] = listget(fields, 39, '0')  # Linux 2.6.18
+        ret['guest_time'] = listget(fields, 40, '0')  # Linux 2.6.24
+        ret['cguest_time'] = listget(fields, 41, '0')  # Linux 2.6.24
 
         return ret
 
@@ -1707,7 +1720,11 @@ class Process(object):
         stime = float(values['stime']) / CLOCK_TICKS
         children_utime = float(values['children_utime']) / CLOCK_TICKS
         children_stime = float(values['children_stime']) / CLOCK_TICKS
-        return _common.pcputimes(utime, stime, children_utime, children_stime)
+        guest_time = float(values['guest_time']) / CLOCK_TICKS
+        cguest_time = float(values['cguest_time']) / CLOCK_TICKS
+        iowait = float(values['delayacct_blkio_ticks']) / CLOCK_TICKS
+        return pcputimes(utime, stime, children_utime, children_stime,
+                         guest_time, cguest_time, iowait)
 
     @wrap_exceptions
     def cpu_num(self):
```

**Comment by giampaolo** [comment]
I confirm this does what intended: it's a counter which keeps track of how much time the CPU waits for blocking I/O to complete. To test this I created a 1G file, I then evicted the page/file cache with `vmtouch` utility, then I ran this script which `read()`s the whole file into memory:

```python
import psutil
sysio1 = psutil.cpu_times().iowait
f = open('filein', 'rb')
f.read()
print(psutil.Process().cpu_times())
sysio2 = psutil.cpu_times().iowait
print(sysio2 - sysio1)
```

It prints:

```
pcputimes(user=0.03, system=0.67, children_user=0.0, children_system=0.0, iowait=0.08)
0.08000000000004093
```

Process iowait and system-wide iowait counters match, so they refer to the same thing. Similarly to system-wide iowait time, this represents an "idle" time, meaning the CPU is not being used. As such the doc should probably mention that `sum(proc.cpu_times())` will not represent the total "busy" time of the process anymore.

We do a similar thing when calculating system-wide CPU percent:
https://github.com/giampaolo/psutil/blob/c10df5aa04e1ced58d19501fa42f08c1b909b83d/psutil/__init__.py#L1775-L1781
...we take into account `iowait` and substract it from the total "busy" CPU time.
